programs/fund-sender/src/lib.rs

```diff
@@ -68,4 +70,34 @@ pub mod fund_sender {
 
         Ok(())
     }
+
+    pub fn store_certificates<'info>(
+        ctx: Context<'_, '_, '_, 'info, StoreCertificates<'info>>,
+    ) -> Result<()> {
+        // send received climate tokens in output_yield_account to a hold account
+        let state = &mut ctx.accounts.state;
+        let output_yield_account = &mut ctx.accounts.output_yield_account;
+        let output_yield_token_account = &mut ctx.accounts.output_yield_token_account;
+        let certificate_vault = &mut ctx.accounts.certificate_vault;
+        if certificate_vault.key() != state.certificate_vault {
+            return Err(ErrorCode::IncorrectHoldAccount.into());
+        }
+
+        let amount: u64 = output_yield_token_account.amount;
+        transfer_token(
+            &state.key(),
+            &AccountsTokenTransfer {
+                source: output_yield_token_account.to_account_info(),
+                dest: certificate_vault.to_account_info(),
+                authority: output_yield_account.to_account_info(),
+            },
+            amount,
+            state.output_yield_account_bump,
+            &state.destination_seed,
+            &ctx.accounts.token_program,
+        )?;
+        state.total_spent += amount;
```

Comment:
Is this correct? If we are defining total_spent as the amount that has gone through the system, then I think this would double-count.

TBH I am not sure we need a total_spent value at all in this program unless you see value in it?

Reply:
Thanks! I have removed this from this instruction. In the program, it should just be a book-keeping of how much sol has been directed to a certain climate project.